### DISCUSSION TITLE
[Service SBS] Private service and repository members on Services

### DISCUSSION THREAD

**Discussion by MRodriguez08** [jhipster/generator-jhipster#24614] [General]
Hello people! I'm a new jhipster developer and I really loved the power and architecture of this framework! I'm still learning all the internals and the de workflows (in particular sbs)
Have a doubt in regards of the Service and Repository Extension, part of the side by side approach.

Why are the services and repository members PRIVATE in the generated classes?? wouldn't be better to have them PROTECTED and be able to use them in the subclasses (ie: UserExtendedService), since they are injected by the super call in constructor.

Right now I have to re-define the services members on my subclass (ext) to be able to use it on some, let's say) overriden method I want to customize.

Pretty sure There's a reason, 
Thanks in advance! great tool here!

**Comment by atomfrede** [comment]
Hi @MRodriguez08 

thanks for bringing this up. I suppose the fields are private as we don't have an option to tell jhipster "this service should be extended" such that the generated class has protected fields only.
Did you read the [tip about combining generated and custom code](https://www.jhipster.tech/tips/035_tip_combine_generation_and_custom_code.html)? 

When it comes to extending repositories you will need to mark the generated repositories with `@NoRepositoryBean`. You could change the fields of generated services to `protected` and use `.yo-resolve` to prevent jhipster from overwriting your changes everytime you update jhipster. Does that help a little?

**Reply by MRodriguez08** [reply]
Thanks for the quick response @atomfrede !!

Will take a look on that doc page. I'm following the SBS approach and extending every class I need to customize. Saw the 2 SBS bible videos (LOL) and, when coding the examples, found that. I wasn't aware of the yo-resolve ignore class option!

In particular, I found the issue on the Rest services layer, So, since you need to inject extended repository classes, makes sense to have them defined as protected to be able to also use them in the Extended Rest Resource (you call the super() passing the extended, the superclass assign that instace to the protected member, it uses internally in non extended method, and allows you to use the service member as a subclass also)
